src/main/resources/templates/db_scripts/oaipmh/createOaiPmhViewFunction.sql

```diff
@@ -0,0 +1,190 @@
+                    jsonb -> 'metadata' ->> 'updatedDate')));
+
+create index if not exists audit_instance_createddate_idx
+    on ${myuniversity}_${mymodule}.audit_instance ((strToTimestamp(
+                jsonb ->> 'createdDate')));
+
+create index if not exists audit_holdings_record_createddate_idx
+    on ${myuniversity}_${mymodule}.audit_holdings_record ((strToTimestamp(
+                jsonb ->> 'createdDate')));
+
+create index if not exists audit_item_createddate_idx
+    on ${myuniversity}_${mymodule}.audit_item ((strToTimestamp(
+                jsonb ->> 'createdDate')));
+
+
+create or replace function ${myuniversity}_${mymodule}.pmh_view_function(startDate timestamp with time zone,
+                                             endDate timestamp with time zone,
+                                             deletedRecordSupport bool default true,
+                                             skipSuppressedFromDiscoveryRecords bool default true)
+    returns table
+            (
+                instanceId          uuid,
+                instanceUpdatedDate timestamp with time zone,
+                deleted             boolean,
+                instanceFields      jsonb
+            )
+as
+$body$
+with instanceIdsInRange as (
+    select inst.id                                              as instanceId,
+           (inst.jsonb #>> '{metadata,updatedDate}')::timestamp as maxDate
+    FROM ${myuniversity}_${mymodule}.instance inst
+    where (inst.jsonb #>> '{metadata,updatedDate}')::timestamp between dateOrMin($1) and dateOrMax($2)
+
+    union all
+    select instanceid,
+           greatest((item.jsonb #>> '{metadata,updatedDate}')::timestamp,
+                    (hr.jsonb #>> '{metadata,updatedDate}')::timestamp) as maxDate
+    from holdings_record hr
+             join ${myuniversity}_${mymodule}.item item
+                  on item.holdingsrecordid = hr.id
+    where ((hr.jsonb #>> '{metadata,updatedDate}')::timestamp between dateOrMin($1) and dateOrMax($2)
+        or
+           (item.jsonb #>> '{metadata,updatedDate}')::timestamp between dateOrMin($1) and dateOrMax($2))
+
+    union all
+    select (audit_holdings_record.jsonb #>> '{record,instanceId}')::uuid,
+           greatest(
+                   (audit_item.jsonb #>> '{record,updatedDate}')::timestamp,
+                   (audit_holdings_record.jsonb #>>
+                    '{record,updatedDate}')::timestamp) as maxDate
+    from audit_holdings_record audit_holdings_record
+             join audit_item audit_item
+                  on (audit_item.jsonb ->> '{record,holdingsRecordId}')::uuid =
+                     audit_holdings_record.id
+    where ((audit_holdings_record.jsonb #>>
+            '{record,updatedDate}')::timestamp between dateOrMin($1) and dateOrMax($2)
+        or
+           (audit_item.jsonb #>> '{record,updatedDate}')::timestamp between dateOrMin($1) and dateOrMax($2))
+),
+     instanceIdsAndDatesInRange as (SELECT instanceId,
```

Comment:
inconsistent use of case... SELECT vs select

Reply:
Corrected, now all keywords are in lower-case.